File: RepoTransBench/target_projects/Python/C++/ai-forever_gigachat/public_tests/test_public_assistants.cpp
#include <gtest/gtest.h>
#include <vector>
#include <string>
#include <map>

TEST(PublicAssistantsTest, AssistantListBehavior) {
    std::vector<std::map<std::string, std::string>> assistants = {
        {{"id", "asst_753"}, {"name", "HelperA"}, {"desc", "Helps with numbers."}},
        {{"id", "asst_111"}, {"name", "HelperB"}, {"desc", "Helps with words."}}
    };
    ASSERT_EQ(assistants[0]["name"], "HelperA");
    ASSERT_TRUE(assistants[1]["desc"].find("Helps with") == 0);
}

TEST(PublicAssistantsTest, AssistantDetailFields) {
    std::map<std::string, std::string> assistant = {
        {"id", "asst_xyz"},
        {"name", "XBot"},
        {"desc", "Handles X-cases"}
    };
    ASSERT_TRUE(assistant.find("id") != assistant.end());
    ASSERT_TRUE(assistant.find("name") != assistant.end());
    ASSERT_EQ(assistant["desc"], "Handles X-cases");
}

File: RepoTransBench/target_projects/Python/Rust/ai-forever_gigachat/public_tests/test_public_assistants.rs
#[test]
fn test_public_assistant_list_behavior() {
    let assistants = vec![
        ("asst_753", "HelperA", "Helps with numbers."),
        ("asst_111", "HelperB", "Helps with words."),
    ];
    assert_eq!(assistants[0].1, "HelperA");
    assert!(assistants[1].2.starts_with("Helps with"));
}

#[test]
fn test_public_assistant_detail_fields() {
    let assistant = ("asst_xyz", "XBot", "Handles X-cases");
    assert_eq!(assistant.0, "asst_xyz");
    assert_eq!(assistant.1, "XBot");
    assert_eq!(assistant.2, "Handles X-cases");
}